source/common/grpc/buffered_async_client_impl.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "source/common/grpc/typed_async_client.h"
+#include "source/common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Grpc {
+
+enum class BufferState { Buffered, Pending };
```

Comment:
Maybe we can call rename Pending to PendingAck or something? There is a slight ambiguity around whether Pending means "pending flush" or "pending response from the server"

Reply:
`PendingAck` is very specific meaning for critical access log. I prefer to use `PendingFlush`.

Reply:
see https://github.com/envoyproxy/envoy/pull/18129